src/etc/one-context.d/loc-10-network.d/netcfg-interfaces

```diff
@@ -0,0 +1,306 @@
+#
+
+gen_iface_conf()
+{
+    cat <<EOT
+iface $DEV inet static
+  address $IP
+  network $NETWORK
+  netmask $MASK
+EOT
+
+    if [ -n "$MTU" ]; then
+        echo "  mtu $MTU"
+    fi
+
+    if [ -n "$GATEWAY" ]; then
+        echo "  gateway $GATEWAY"
+
+        if [ -n "$METRIC" ]; then
+            echo "  metric $METRIC"
+        fi
+    fi
+
+    echo ""
+}
+
+gen_alias_conf()
+{
+    cat <<EOT
+iface $DEV inet static
+  address $IP
+  network $NETWORK
+  netmask $MASK
+EOT
+
+echo ""
+}
+
+gen_alias6_conf()
+{
+    case "${OS_ID}" in
+        alpine)
+            cat <<EOT
+iface $DEV inet6 static
+  address $IP6
+  netmask ${IP6_PREFIX_LENGTH:-64}
+  pre-up echo 0 > /proc/sys/net/ipv6/conf/${DEV}/autoconf
+  pre-up echo 0 > /proc/sys/net/ipv6/conf/${DEV}/accept_ra
+EOT
+            ;;
+        debian|ubuntu|devuan)
+            cat <<EOT
+iface $DEV inet6 static
+  address $IP6
+  netmask ${IP6_PREFIX_LENGTH:-64}
+  autoconf 0
+  accept_ra 0
+EOT
+            ;;
+    esac
+
+    if [ -n "$IP6_ULA" ]; then
+        cat <<EOT
+
+iface $DEV inet6 static
+  address $IP6_ULA
+  netmask 64
+EOT
+    fi
+
+    echo ""
+}
+
+gen_iface6_conf()
+{
+    case "${OS_ID}" in
+        alpine)
+            cat <<EOT
+iface $DEV inet6 static
+  address $IP6
+  netmask ${IP6_PREFIX_LENGTH:-64}
+  pre-up echo 0 > /proc/sys/net/ipv6/conf/${DEV}/autoconf
+  pre-up echo 0 > /proc/sys/net/ipv6/conf/${DEV}/accept_ra
+EOT
+            ;;
+        debian|ubuntu|devuan)
+            cat <<EOT
+iface $DEV inet6 static
+  address $IP6
+  netmask ${IP6_PREFIX_LENGTH:-64}
+  autoconf 0
+  accept_ra 0
+EOT
+            ;;
+    esac
+
+    if [ -n "$MTU" ]; then
+        echo "  mtu $MTU"
+    fi
+
+    if [ -n "$GATEWAY6" ]; then
+        echo "  gateway $GATEWAY6"
+    fi
+
+    if [ -n "$IP6_ULA" ]; then
+        cat <<EOT
+
+iface $DEV inet6 static
+  address $IP6_ULA
+  netmask 64
+EOT
+
+        case "${OS_ID}" in
+            debian|ubuntu|devuan)
+                cat <<EOT
+  autoconf 0
+  accept_ra 0
+EOT
+                ;;
+        esac
+
+        if [ -n "$MTU" ]; then
+            echo "  mtu $MTU"
+        fi
+    fi
+
+    echo ""
+}
+
+gen_network_configuration()
+{
+    cat <<EOT
+auto lo
+iface lo inet loopback
+
+EOT
+
+    INTERFACE_MAC=$(get_interface_mac)
+    CONTEXT_INTERFACES=$(get_context_interfaces)
+
+    for iface in $CONTEXT_INTERFACES; do
+        MAC=$(get_iface_var "$iface" "MAC")
+        DEV=$(get_dev "$INTERFACE_MAC" "$MAC")
+
+        IP=$(get_ip "$iface")
+        NETWORK=$(get_network "$iface")
+        MASK=$(get_mask "$iface")
+        MTU=$(get_mtu "$iface")
+        GATEWAY=$(get_gateway "$iface")
+        METRIC=$(get_iface_var "$iface" "METRIC")
+
+        IP6=$(get_iface_var "$iface" "IP6")
+        [ -z "$IP6" ] && IP6=$(get_iface_var "$iface" "IPV6")
+        IP6_PREFIX_LENGTH=$(get_iface_var "$iface" "IP6_PREFIX_LENGTH")
+        IP6_ULA=$(get_iface_var "$iface" "IP6_ULA")
+        GATEWAY6=$(get_gateway6 "$iface")
```

Comment:
This is redundant in all scripts, maybe we could:
- get rid that and use directly those functions OR
- have a dedicated function, which sets these parameters?